src/App.tsx

```diff
@@ -15,36 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_ALPHABETICALLY = 'name',
+  SORT_BY_LENGTH = 'length',
+  none = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverse: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((nameA, nameB) => {
+      switch (sortField) {
+        case SortType.SORT_ALPHABETICALLY:
+          return nameA.localeCompare(nameB);
+
+        case SortType.SORT_BY_LENGTH:
+          return nameA.length - nameB.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.none);
+  const [reverse, setReverse] = useState(false);
```

Comment:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
```suggestion
  const [isReversed, setIsReverse] = useState(false);
```